evm/src/executor/fork/backend.rs

```diff
@@ -249,7 +256,10 @@ where
                         trace!(target: "backendhandler", "last sender dropped, ready to drop (&flush cache)");
                         return Poll::Ready(())
                     }
-                    _ => break,
+                    Poll::Pending => {
+                        cx.waker().wake_by_ref();
```

Comment:
i'm confused, what does it do?

Reply:
I'm not sure why, but in node fork mode, the task did not wake up after reaching this state, even if new messages were sent to the channel...

this explicitly tells the scheduler to poll it again

Reply:
is there an easy way to reproduce it? there is definitely something fishy with spawning and hyper, but I processed ~8 months worth of blocks without this change.

just tested it and this single line cuts the throughput in half.

Reply:
because of that (probably) hyper problem i'm using foundry-evm with `Ws` provider & most of the processing lives in `futures::executor::ThreadPool::spawn_with_handle`, so maybe perf degradation is not visible in foundry.

Reply:
yeh this a rather poor solution and turns this basically into a busy loop. 
performance for foundry itself is not really an issue here, so because I wasn't able to track this down made this change.

> is there an easy way to reproduce it? 

I believe it was this test or any other fork test 
https://github.com/foundry-rs/foundry/blob/master/anvil/tests/it/fork.rs#L34


just tested again and it hangs when the SharedBackend sends the request if I remove the wake, not sure why though

Reply:
@univerz realize I'm hijacking the thread but if you wanna join the foundry telegram group to have more synchronous conversations, it'd be great! https://t.me/foundry_rs